### PR TITLE
Fix for Geo Location Data in Ad

### PR DESCRIPTION
**Pull request by rybodiddly**
Fixes an issue where ads are given 0,0 for geo location data, resulting in ads appearing in the South Atlantic.
Requires an additional module installation: `pgeocode`

### CODE REVIEW

File: kijiji_manager/kijijiapi.py [hunk 1]
```diff
@@ -381,7 +382,25 @@ def post_conversation_reply(self, user_id, token, conversation_id, ad_id, userna
             return doc
         else:
             raise KijijiApiException(self._error_reason(doc))
-
+    
+    def geo_location(self, postal_code):
+        postalcode = self.check_postal_code_length(postal_code)
```
Comment (jackm) [inline kijiji_manager/kijijiapi.py:387]:
```suggestion
        # pgeocode.Nominatim.query_postal_code actually only needs the first three characters of the postal code to do the lookup
        postalcode = postal_code[:3]
```

`pgeocode.Nominatim.query_postal_code` actually only needs the first three characters of the postal code to do the lookup. The latter 3 postal code characters do not improve the accuracy of the result.

File: kijiji_manager/kijijiapi.py [hunk 2]
```diff
@@ -381,7 +382,25 @@ def post_conversation_reply(self, user_id, token, conversation_id, ad_id, userna
...
+        else:
+            return location
+        
+    def check_postal_code_length(self, postal_code):
+        if len(postal_code) == 6:
+            section1 = postal_code[:3]
+            section2 = postal_code[3:6]
+            return section1 + ' ' + section2
+        else:
+            return postal_code
```
Comment (jackm) [inline kijiji_manager/kijijiapi.py:402]:
```suggestion
```

Postal code is already validated via the form input. This isn't neccessary.

File: kijiji_manager/kijijiapi.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 from xml.parsers.expat import ExpatError, errors
 
 import httpx
+import pgeocode
```
Comment (jackm) [inline kijiji_manager/kijijiapi.py:4]: Package should also be added to requirements file

**Review by jackm — CHANGES_REQUESTED** [review]

File: kijiji_manager/kijijiapi.py (hunk 1 above)
Comment (jackm) [reply kijiji_manager/kijijiapi.py:387]: Alternatively we could change the postal code validation on the form input to enforce having a space between the first 3 characters and the last 3 characters. Ensuring that there is a space means you can pass the postal code string directly to `query_postal_code` without having to modify it (it splits the two components on whitespace and uses only the first component).

Comment (rybodiddly) [reply kijiji_manager/kijijiapi.py:387]: The postal code splitting was carried over from experiments with geopy (which was pickier) prior to the switch to pgeocode. Just ran a test with pgeocode and you can feed it just the first 3 characters, all 6 or six with a space. It'll give you the same results regardless. So yes, completely agree the postal code splitting is unnecessary.

**Review by jackm — APPROVED** [review]

**Pull request merged**